### ISSUE TITLE
Minor inconsistencies in Simulation object

### ISSUE THREAD

**Issue by unoebauer** [tardis-sn/tardis#442]
While working on PR #440, I noticed that some simulation properties are accessed within if statements in the legacy runner but are never set before (I've already added a comment):

``` python
            elif not converged and self.converged:
                # UMN Warning: the following two iterations attributes of the Simulation object don't exist
                self.iterations_remaining = self.iterations_max_requested - self.iterations_executed
                self.converged = False
```

Any reason for this? While running simple tardis setups (mostly tardis_example), the calculation never enters this segment of the if statement. Still, this should be resolved.... (more of these inconsistencies may be lurking around the simulation/base.py file)

**Comment by souravsingh** [comment]
I would like to work on the issue.

**Comment by iamgroot42** [comment]
The properties `self.iterations_max_requested` and `self.iterations_executed` are neither initialized nor used/changed anywhere else. Why does do they even exist ?

**Comment by unoebauer** [comment]
Hi @ftsamis - have you done anything about this while checking out the convergence strategies?

**Comment by ftsamis** [comment]
This code has been completely rewritten as part of #652, so we should close this issue after the PR gets merged.

**Comment by ftsamis** [comment]
Fixed as part of #652.
